src/mesido/electricity_physics_mixin.py

```diff
@@ -114,6 +119,26 @@ def pre(self):
                 self.__asset_is_switched_on_var[var_name] = ca.MX.sym(var_name)
                 self.__asset_is_switched_on_bounds[var_name] = (0.0, 1.0)
 
+        if options["electrolyzer_efficiency"] == ElectrolyzerOption.LINEARIZED_THREE_LINES_EQUALITY:
+            n_lines = 3
+            for n_line in range(n_lines):
+                self.__electrolyzer_is_active_linear_segment_map[f"line_{n_line}"] = {}
+                self.__electrolyzer_is_active_linear_segment_var[f"line_{n_line}"] = {}
+                self.__electrolyzer_is_active_linear_segment_bounds[f"line_{n_line}"] = {}
+                for asset in [
+                    *self.energy_system_components.get("electrolyzer", []),
+                ]:
+                    var_name = f"{asset}__line_{n_line}_active"
+                    self.__electrolyzer_is_active_linear_segment_map[f"line_{n_line}"][
+                        asset
+                    ] = var_name
+                    self.__electrolyzer_is_active_linear_segment_var[f"line_{n_line}"][var_name] = (
+                        ca.MX.sym(var_name)
+                    )
+                    self.__electrolyzer_is_active_linear_segment_bounds[f"line_{n_line}"][
+                        var_name
+                    ] = (0.0, 1.0)
```

Comment:
Another recommendation, the map from asset to variable names is fine that have it nested [asset][n_line] = var_name, but then the __electrolyzer_is_active_linear_segment_var does not have to be a nested dic, but can solely be as all variable names are already unique and linked with the map above.
 self.__electrolyzer_is_active_linear_segment_var[var_name]= ca.MX.sym(var_name)
 
 Same would then also hold for the bounds ofcourse.

Reply:
Changed